lib/createFormatter.js

```diff
@@ -3,15 +3,17 @@ const winston = require('winston')
 const truncate = require('./truncate')
 
 module.exports = function createFormatter(newrelic) {
-  const jsonFormatter = winston.format.json()
-  return winston.format((info, opts) => {
-    const metadata = newrelic.getLinkingMetadata(true)
+  // Stub API means agent is not enabled.
+  if (!newrelic.shim) {
```

Comment:
Looks like we check to see if the agent is enabled when the formatter is created. Is it possible for the agent to be enabled when this formatter is created, but then somehow disabled later?

Reply:
No, we currently only check that configuration once upon initial load.

If we did make that more dynamic somehow... we'd likely swap out API's under the hood and we need something safe for `shim` anyways which would likely be swapping out API implementations behind the scenes.